Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -57,83 +57,132 @@ createCoordinateSystem(const FVector& center, double scale) {
            ACesiumGeoreference::StaticClass(),
            flags);
        actorIterator;
        ++actorIterator) {
     AActor* actor = *actorIterator;
-    if (actor->GetLevel() == world->PersistentLevel &&
-        actor->ActorHasTag(DEFAULT_GEOREFERENCE_TAG)) {
-      pGeoreference = Cast<ACesiumGeoreference>(actor);
+    if (actor->GetLevel() == World->PersistentLevel &&
+        actor->ActorHasTag(Tag)) {
+      Georeference = Cast<ACesiumGeoreference>(actor);
       break;
     }
   }
-  if (!pGeoreference) {
-    // Legacy method of finding Georeference, for backwards compatibility with
-    // existing projects
-    ACesiumGeoreference* pGeoreferenceCandidate =
-        FindObject<ACesiumGeoreference>(
-            world->PersistentLevel,
-            TEXT("CesiumGeoreferenceDefault"));
-    // Test if PendingKill
-    if (IsValid(pGeoreferenceCandidate)) {
-      pGeoreference = pGeoreferenceCandidate;
-    }
-  }
-  if (!pGeoreference) {
-    UE_LOG(
-        LogCesium,
-        Verbose,
-        TEXT("Creating default Georeference for actor %s"),
-        *WorldContextObject->GetName());
-    // Spawn georeference in the persistent level
-    FActorSpawnParameters spawnParameters;
-    spawnParameters.SpawnCollisionHandlingOverride =
-        ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
-    spawnParameters.OverrideLevel = world->PersistentLevel;
-    pGeoreference = world->SpawnActor<ACesiumGeoreference>(spawnParameters);
-    // Null check so the editor doesn't crash when it makes arbitrary calls to
-    // this function without a valid world context object.
-    if (pGeoreference) {
-      pGeoreference->Tags.Add(DEFAULT_GEOREFERENCE_TAG);
-    }
 
-  } else {
-    UE_LOG(
-        LogCesium,
-        Verbose,
-        TEXT("Using existing Georeference %s for actor %s"),
-        *pGeoreference->GetName(),
-        *WorldContextObject->GetName());
+  return Georeference;
+}
+
+ACesiumGeoreference*
+FindGeoreferenceWithDefaultName(const UObject* WorldContextObject) {
+  UWorld* World = WorldContextObject->GetWorld();
+  if (!IsValid(World))
+    return nullptr;
+
+  ACesiumGeoreference* Candidate = FindObject<ACesiumGeoreference>(
+      World->PersistentLevel,
+      TEXT("CesiumGeoreferenceDefault"));
+
+  // Test if PendingKill
+  if (IsValid(Candidate)) {
+    return Candidate;
   }
-  return pGeoreference;
+
+  return nullptr;
+}
+
+ACesiumGeoreference*
+CreateDefaultGeoreference(const UObject* WorldContextObject, const FName& Tag) {
+  UWorld* World = WorldContextObject->GetWorld();
+  if (!IsValid(World))
+    return nullptr;
+
+  // Spawn georeference in the persistent level
+  FActorSpawnParameters spawnParameters;
+  spawnParameters.SpawnCollisionHandlingOverride =
+      ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+  spawnParameters.OverrideLevel = World->PersistentLevel;
+
+  ACesiumGeoreference* Georeference =
+      World->SpawnActor<ACesiumGeoreference>(spawnParameters);
+  Georeference->Tags.Add(Tag);
```

Comment:
Looks like you removed the null check against the returned pointer from SpawnActor.

That's fine, but if you're convinced this should always be non-null, might add an assert for that.